osu.Game/Screens/Play/HUD/ClicksPerSecond/ClicksPerSecondCounter.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Localisation;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Skinning;
+using osuTK;
+
+namespace osu.Game.Screens.Play.HUD.ClicksPerSecond
+{
+    public class ClicksPerSecondCounter : RollingCounter<int>, ISkinnableDrawable
+    {
+        private const float alpha_when_invalid = 0.3f;
+
+        private readonly Bindable<bool> valid = new Bindable<bool>();
+
+        private ClicksPerSecondCalculator? calculator;
```

Comment:
```suggestion
        [Resolved(canBeNull: true)]
        private ClicksPerSecondCalculator? calculator;
```

and remove the value transfer in `load()`.

Reply:
Hmm... But then it would show up in the skin editor at song select afaik, which doesn't really make sense, but I agree that it's better to use `ResolvedAttribute` for this. Is there a way to force the counter to only show in a Gameplay context using the `ResolvedAttribute`?

Reply:
the `canBeNull` makes it optional / nullable. Moving this won't change the actual logic at all.